src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
+   * child component's element from this component's element
+   *
+   * @param  {vjs.Component} component Component to remove
+   */
+  removeChild(component){
+    if (typeof component === 'string') {
+      component = this.getChild(component);
+    }
 
-  // If a name wasn't used to create the component, check if we can use the
-  // name function of the component
-  componentName = componentName || (component.name && component.name());
+    if (!component || !this.children_) return;
 
-  if (componentName) {
-    this.childNameIndex_[componentName] = component;
-  }
+    var childFound = false;
+    for (var i = this.children_.length - 1; i >= 0; i--) {
+      if (this.children_[i] === component) {
+        childFound = true;
+        this.children_.splice(i,1);
+        break;
+      }
+    }
 
-  // Add the UI object's element to the container div (box)
-  // Having an element is not required
-  if (typeof component['el'] === 'function' && component['el']()) {
-    this.contentEl().appendChild(component['el']());
-  }
+    if (!childFound) return;
 
-  // Return so it can stored on parent object if desired.
-  return component;
-};
+    this.childIndex_[component.id()] = null;
+    this.childNameIndex_[component.name()] = null;
 
-/**
- * Remove a child component from this component's list of children, and the
- * child component's element from this component's element
- *
- * @param  {vjs.Component} component Component to remove
- */
-Component.prototype.removeChild = function(component){
-  if (typeof component === 'string') {
-    component = this.getChild(component);
+    var compEl = component.el();
+    if (compEl && compEl.parentNode === this.contentEl()) {
+      this.contentEl().removeChild(component.el());
+    }
   }
 
-  if (!component || !this.children_) return;
+  /**
+   * Add and initialize default child components from options
+   *
+   *     // when an instance of MyComponent is created, all children in options
+   *     // will be added to the instance by their name strings and options
+   *     MyComponent.prototype.options_.children = {
+   *       myChildComponent: {
+   *         myChildOption: true
+   *       }
+   *     }
+   *
+   *     // Or when creating the component
+   *     var myComp = new MyComponent(player, {
+   *       children: {
+   *         myChildComponent: {
+   *           myChildOption: true
+   *         }
+   *       }
+   *     });
+   *
+   * The children option can also be an Array of child names or
+   * child options objects (that also include a 'name' key).
+   *
+   *     var myComp = new MyComponent(player, {
+   *       children: [
+   *         'button',
+   *         {
+   *           name: 'button',
+   *           someOtherOption: true
+   *         }
+   *       ]
+   *     });
+   *
+   */
+  initChildren(){
+    let parent = this;
+    let parentOptions = parent.options();
+    let children = parentOptions['children'];
+
+    let handleAdd;
```

Comment:
It made sense when it was vars. The if block used children which was defined with parentOptions which was defined with parent. I didn't actually touch this part but I've cleaned it up for you. :)